File: finstack-py/tests/golden/runners/pricing_inflation_linked_bond.py
"""Domain runner for `fixed_income.inflation_linked_bond` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one inflation-linked bond fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_inflation_linked_bond.rs
//! Domain runner for `fixed_income.inflation_linked_bond` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Inflation-linked bond pricing golden runner.
pub struct InflationLinkedBondRunner;

impl DomainRunner for InflationLinkedBondRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
